src/tsdb.c

```diff
@@ -456,3 +571,40 @@ int SeriesDeleteSrcRule(Series *series, RedisModuleString *srctKey) {
     }
     return FALSE;
 }
+
+/*
+ * This function calculate aggregation value of a range.
+ *
+ * If `val` is NULL, the function will update the context of `rule`.
+ */
+int SeriesCalcRange(Series *series,
```

Comment:
Rule is a must. Just like series.
It could make it clearer if we `assert` must arguments.

Whether `*val` is null or not is not straight forward which is why I left a comment.